src/main/java/com/cloudbees/jenkins/GitHubPushCause.java

```diff
@@ -34,4 +35,21 @@ public String getShortDescription() {
         String pusher = pushedBy != null ? pushedBy : "";
         return "Started by GitHub push by " + pusher;
     }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(this.pushedBy);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!getClass().isInstance(obj)) {
```

Comment:
`GitHubPushCause` is not final and this method is not reflective and could incorrectly match any subclass of this type who adds new fields and overrideds `equals()`
so `a.equals(b) != b.equals(a)`